api/task/ingest.go

```diff
@@ -635,3 +637,52 @@ func getUniqueDatasetName(meta *model.Metadata, storage api.MetadataStorage) (st
 
 	return getUniqueString(meta.Name, datasetNames), nil
 }
+
+func verifyData(pg *postgres.Database, meta *model.Metadata, storage api.MetadataStorage, tableName string) {
+	validTypes := postgres.GetValidTypes()
+	ds, err:=storage.FetchDataset(meta.Name, false, true)
+	if err != nil{
+		return
+	}
+	//removing double and geometry for now
+	double:="double precision"
+	geometry:="geometry"
+	mainValidTypes := []string{}
+	provenance:="postgres-valid"
+	for i := range validTypes{
+		if validTypes[i] != double && validTypes[i] != geometry{
+			mainValidTypes = append(mainValidTypes, validTypes[i])
+		}
+	}
+	// if view can succeed on column add potential type to column
+	for i := range ds.Variables{
+		suggestedMap := make(map[string]bool)
+		for t:=range ds.Variables[i].SuggestedTypes{
+			suggestedMap[ds.Variables[i].SuggestedTypes[t].Type]=true
+		}
+		for j := range mainValidTypes{
+			isValid, err := pg.IsColumnType(tableName, ds.Variables[i], mainValidTypes[j])
+			if err != nil{
+				continue
+			}
+			if isValid {
+				d3mTypes,err:=postgres.MapPostgresTypeToD3MType(mainValidTypes[j])
+				if err != nil{
+					continue
+				}
+				for k := range d3mTypes{
+					// this could be moved up to an exit case above but a lot of the upconversion from pg to d3m types involves multiple results
+					if suggestedMap[d3mTypes[k]] {
+						continue
+					}
+					suggestedType := model.SuggestedType{Probability:0, Type:d3mTypes[k], Provenance:provenance}
+					ds.Variables[i].SuggestedTypes = append(ds.Variables[i].SuggestedTypes, &suggestedType)
+				}
+			}
+		}
+	}
+	// save changes
+	storage.UpdateDataset(ds)
+	
+	return
```

Comment:
As general comment, this function needs to add error handling in some way. If we go forward with the assumption that suggested types has the complete list, then this needs to be considered an integral part of the ingest process so any failure here needs to fail the ingest.

The function should probably have error as return type, and whenever something errors it should forward that error.